src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```
@@ -0,0 +1,178 @@
+
+    /**
+     * Instantiates a new CreateIngestPipelineStep
+     *
+     * @param client The client to create a pipeline and store workflow data into the global context index
+     */
+    public CreateIngestPipelineStep(Client client) {
+        this.clusterAdminClient = client.admin().cluster();
+        this.client = client;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+
+        CompletableFuture<WorkflowData> createIngestPipelineFuture = new CompletableFuture<>();
+
+        String pipelineId = null;
+        String description = null;
+        String modelId = null;
+        String inputFieldName = null;
+        String outputFieldName = null;
+        BytesReference configuration = null;
+
+        // Extract required content from workflow data and generate the ingest pipeline configuration
+        for (WorkflowData workflowData : data) {
+
+            Map<String, String> parameters = workflowData.getParams();
+            Map<String, Object> content = workflowData.getContent();
+            logger.debug("Previous step sent params: {}, content: {}", parameters, content);
+
+            if (content.containsKey(PIPELINE_ID_FIELD)) {
+                pipelineId = (String) content.get(PIPELINE_ID_FIELD);
+            }
+            if (content.containsKey(DESCRIPTION_FIELD)) {
+                description = (String) content.get(DESCRIPTION_FIELD);
+            }
+            if (content.containsKey(MODEL_ID_FIELD)) {
+                modelId = (String) content.get(MODEL_ID_FIELD);
+            }
+            if (content.containsKey(INPUT_FIELD)) {
+                inputFieldName = (String) content.get(INPUT_FIELD);
+            }
+            if (content.containsKey(OUTPUT_FIELD)) {
+                outputFieldName = (String) content.get(OUTPUT_FIELD);
+            }
+
+            if (Stream.of(pipelineId, description, modelId, inputFieldName, outputFieldName).allMatch(x -> x != null)) {
```

I've never seen this for null checking, I like it!  

However, I don't see an "else" clause here, should we throw an exception if we don't have all the required fields?

---

The exception is thrown below when we check if the configuration is null or not, but ill move this around so its more apparent